acorn/src/expression.js

```diff
@@ -431,13 +421,34 @@ pp.parseExprAtom = function(refDestructuringErrors) {
   }
 }
 
-pp.parseDynamicImport = function() {
+pp.parseExprImport = function() {
   const node = this.startNode()
-  this.next()
-  if (this.type !== tt.parenL) {
+  this.next() // skip `import`
+  switch (this.type) {
+  case tt.parenL:
+    return this.parseDynamicImport(node)
+  default:
     this.unexpected()
   }
-  return this.finishNode(node, "Import")
+}
+
+pp.parseDynamicImport = function(node) {
+  this.next() // skip `(`
+
+  // Parse node.source.
+  node.source = this.parseMaybeAssign()
+
+  // Verify ending.
```

Comment:
Thanks for adjusting the ellipses thing. Any reason the 8 lines below can't simply be `this.expect(tt.parenR)`?

Reply:
I wanted to show a detailed message for trailing commas because `import(s)` is similar to call expressions, but doesn't support trailing commas. I.e., it handles `import(s,)` as special, but not `import(a,b)` and `import(s!`.

Reply:
I have moved the logic to the errored code path.